base/abstractarray.jl

```diff
@@ -3525,13 +3525,45 @@ julia> map(+, [1 2; 3 4], [1,10,100,1000], zeros(3,1))  # iterates until 3rd is
 """
 map(f, it, iters...) = collect(Generator(f, it, iters...))
 
+# Generic versions of push! for AbstractVector
+# These are specialized further for Vector for faster resizing and setindexing
+function push!(a::AbstractVector{T}, item) where T
+    # convert first so we don't grow the array if the assignment won't work
+    itemT = item isa T ? item : convert(T, item)::T
+    new_length = length(a) + 1
+    resize!(a, new_length)
+    a[new_length] = itemT
+    return a
+end
+
+# specialize and optimize the single argument case
+function push!(a::AbstractVector{Any}, @nospecialize x)
+    new_length = length(a) + 1
+    resize!(a, new_length)
+    a[new_length] = x
+    return a
+end
+function push!(a::AbstractVector{Any}, @nospecialize x...)
+    @_terminates_locally_meta
+    na = length(a)
+    nx = length(x)
+    resize!(a, na + nx)
+    for i = 1:nx
+        a[na+i] = x[i]
+    end
+    return a
+end
```

Comment:
This reverts the fix for https://github.com/JuliaLang/julia/issues/15868, so is probably a no-go. The other functions are fine.
```suggestion
```

Reply:
This is the exact analog of the `Vector{Any}` case:
https://github.com/JuliaLang/julia/blob/master/base%2Farray.jl#L1278-L1287

Could you clarify how this reverts the fix but the above code that I cited does not?

Note that this is for `AbstractVector{Any}`, not `AbstractVector{<: Any}`.

Reply:
Oh, I see. Yes, this looks fine too then